### PR TITLE
olwa-pr

### PR DESCRIPTION
**Pull request by oliver-walker-hms**

### CODE REVIEW

**Comment by TomKimsey** [comment]
i feel left out

**Comment by oliver-walker-hms** [comment]
there we go

File: src/com/hms/flexyosisoftconnector/PayloadBuilder.java [hunk 1]
```diff
@@ -298,15 +298,25 @@ public static String getIntegerTypeBody() {
...
+   * @param format specifies what format the type uses on the AF server for bit count and signed or
+   *     unsigned
    * @return returns the JSON payload to construct a new Type message.
    */
-  private static String getGenericTypeBody(String type) {
+  private static String getGenericTypeBody(String type, String format) {
+    String formatString = "";
+
+    // include if not booleans or strings
+    if (type.compareToIgnoreCase("string") != 0 || type.compareToIgnoreCase("boolean") != 0) {
```
Comment (TomKimsey) [inline src/com/hms/flexyosisoftconnector/PayloadBuilder.java:316]: Couldn't this just be
```suggestion
    if (type.compareToIgnoreCase("string") || type.compareToIgnoreCase("boolean")) {
```

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
 # Flexy OSIsoft Changelog
 
+## v2.4
+### Major Changes
++ Fix integer type storage bug in OSIsoft AF server
+### Minor Changes
++ refactored code
```
Comment (TomKimsey) [inline docs/CHANGELOG.md:7]:
```suggestion
```

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,12 @@
 # Flexy OSIsoft Changelog
 
+## v2.4
+### Major Changes
++ Fix integer type storage bug in OSIsoft AF server
+### Minor Changes
++ refactored code
++ specify type format for OMF instead of using default
```
Comment (TomKimsey) [inline docs/CHANGELOG.md:8]:
```suggestion
+ Specified type format for OMF instead of using default type
```

File: docs/CHANGELOG.md (hunk 2 above)
Comment (TomKimsey) [reply docs/CHANGELOG.md:7]: Dont think this really needs to be in the changelog

File: docs/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,12 @@
 # Flexy OSIsoft Changelog
 
+## v2.4
+### Major Changes
++ Fix integer type storage bug in OSIsoft AF server
```
Comment (TomKimsey) [inline docs/CHANGELOG.md:5]:
```suggestion
+ Fixed integer type storage bug in OSIsoft AF server
```

**Comment by oliver-walker-hms** [comment]
oh wait, need to re-push. the code style checks are complaining

**Comment by oliver-walker-hms** [comment]
fixed it, it's good for review now

File: docs/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,11 @@
 # Flexy OSIsoft Changelog
 
+## v2.4
+### Major Changes
++ Fixed integer type storage bug in OSIsoft AF server
+### Minor Changes
++ specified type format for OMF instead of using default type
```
Comment (TomKimsey) [inline docs/CHANGELOG.md:7]:
```suggestion
+ Specified type format for OMF instead of using default type
```

**Pull request merged**